### DISCUSSION TITLE
Pure random when estimating V(s), but epsilon_greedy when estimating Q(s,a)?

### DISCUSSION THREAD

**Discussion by QuantHao** [PacktPublishing/Tensorflow-2-Reinforcement-Learning-Cookbook#26] [Reinforcement Learning] [answered]
Hi, @praveen-palanisamy 

I confused about when using pure random when estimating V(s) at https://github.com/PacktPublishing/Tensorflow-2-Reinforcement-Learning-Cookbook/blob/101f9866f6105b0ac696f9881d475b8a525eb8ea/Chapter02/4_monte_carlo_prediction_and_control_rl.py#L26

But using epsilon_greedy when estimating Q(s,a) at https://github.com/PacktPublishing/Tensorflow-2-Reinforcement-Learning-Cookbook/blob/101f9866f6105b0ac696f9881d475b8a525eb8ea/Chapter02/4_monte_carlo_prediction_and_control_rl.py#L77-L78

Is there any reason NOT using epsilon_greedy to estimate V(s)? Thanks.

**Comment by praveen-palanisamy** [comment, marked as answer]
Hi @QuantHao, Good to hear from you again!
There's no reason **not** to use the `epsilon_greedy` policy to estimate ![V_{\pi}(s)](https://latex.codecogs.com/gif.latex?V_{\pi}(s)).
You can estimate the state-value function for the `epsilon_greedy` policy and even compare it with the state-value function of the `random` policy. Similarly for the action-value function.

**Reply by QuantHao** [reply]
Thanks so much for answering. 

Your book is so far the BEST book teaching about using TF2 in RL from scratch!! Really appreciate your work.

**Reply by praveen-palanisamy** [reply]
That's nice to hear and thank your for your appreciation!
Please continue to learn, build and create!
Also, report the bugs you find :). It's great to see your attention to the details and the spot-on issues you have created on this repo. :clap:

**Comment by praveen-palanisamy** [comment]
Converting this to a Q&A Discussion on RL to organize for the community's benefit.
